app/src/main/kotlin/com/rockspin/bargainbits/ui/deals/DealListActivity.kt

```diff
@@ -0,0 +1,136 @@
+package com.rockspin.bargainbits.ui.deals
+
+import android.app.SearchManager
+import android.arch.lifecycle.ViewModelProviders
+import android.content.Context
+import android.content.Intent
+import android.databinding.DataBindingUtil
+import android.os.Bundle
+import android.support.design.widget.Snackbar
+import android.support.v4.view.MenuItemCompat
+import android.support.v7.widget.SearchView
+import com.jakewharton.rxbinding2.support.v7.widget.itemClicks
+import com.jakewharton.rxbinding2.widget.itemSelections
+import com.rockspin.bargainbits.R
+import com.rockspin.bargainbits.databinding.ActivityDealListBinding
+import com.rockspin.bargainbits.ui.BaseActivity
+import com.rockspin.bargainbits.ui.store_filter.StoreFilterDialogFragment
+import com.rockspin.bargainbits.ui.watch_list.WatchListActivity
+import com.rockspin.bargainbits.util.visible
+import com.rockspin.bargainbits.utils.Feedback
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.rxkotlin.addTo
+import io.reactivex.rxkotlin.merge
+import io.reactivex.subjects.PublishSubject
+import org.codechimp.apprater.AppRater
+import javax.inject.Inject
+
+class DealListActivity : BaseActivity(), StoreFilterDialogFragment.OnDialogClosedListener {
+
+    @Inject lateinit var adapter: DealListAdapter
+    @Inject lateinit var factory: DealListViewModelFactory
+
+    private var noInternetSnackbar: Snackbar? = null
+    private val onDestroyDisposable = CompositeDisposable()
+    private val onStoreFilterClosed = PublishSubject.create<Unit>()
+
+    private val binding by lazy { DataBindingUtil.setContentView<ActivityDealListBinding>(this, R.layout.activity_deal_list) }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        // TODO - remember user default choice for deal sorting
```

Comment:
Another TODO here that will still be here in six months if it isn't tracked properly.